llvm/include/llvm/ADT/APInt.h

```diff
@@ -2193,6 +2193,12 @@ inline const APInt absdiff(const APInt &A, const APInt &B) {
   return A.uge(B) ? (A - B) : (B - A);
 }
 
+/// Compute the higher order bits of unsigned multiplication of two APInts
+APInt mulhu(const APInt &C1, const APInt &C2);
+
+/// Compute the higher order bits of signed multiplication of two APInts
```

Comment:
Also here. The signed case could also use an example with negative numbers

Reply:
You mean in the comment? I have given an example for this in APIntTest.cpp.

Reply:
Yes, in this context the comment above the function declaration is the documentation.

Reply:
Do let me know if the comments are explanatory enough now.

Reply:
Could you also add a formula similar to the one above? The main thing is to explain what happens with the sign bit IMO.